catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/stage/MultipleBroadcastsTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.stage;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.content.BroadcastHandler;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.SingleSprite;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.BroadcastBrick;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.formulaeditor.datacontainer.DataContainer;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.ProjectSetupUtils;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(AndroidJUnit4.class)
+public class MultipleBroadcastsTest {
+
+	private static final String BROADCAST_MESSAGE_1 = "message1";
+
+	@Rule
+	public BaseActivityInstrumentationRule<StageActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(StageActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		BroadcastHandler.clearActionMaps();
+	}
+
+	@Test
+	public void testSendMultipleBroadcasts() throws InterruptedException {
+
+		int[] values = new int[] {30, 60, -30, -60};
+		List<UserVariable> variables = createProject(values);
+
+		baseActivityTestRule.launchActivity(null);
+		Thread.sleep(1500);
+
+		for (int i = 0; i < variables.size(); i++) {
+			UiTestUtils.waitForVariableValueEquals(variables.get(i), values[i], 2000);
+		}
+	}
+
+	private List<UserVariable> createProject(int[] values) {
+		Project project = UiTestUtils.createEmptyProject("test");
```

Comment:
it would probably be nicer to have your own createEmptyProject function here, where you already get the data container so you dont have to dig for it.